substrate/client/transaction-pool/tests/fatp_prios.rs

```diff
@@ -247,3 +249,315 @@ fn fatp_prio_watcher_future_lower_prio_gets_dropped_from_all_views() {
 	assert_ready_iterator!(header01.hash(), pool, [xt2, xt1]);
 	assert_ready_iterator!(header02.hash(), pool, [xt2, xt1]);
 }
+
+#[test]
+fn fatp_prios_watcher_full_mempool_higher_prio_is_accepted() {
+	sp_tracing::try_init_simple();
+
+	let builder = TestPoolBuilder::new();
+	let (pool, api, _) = builder.with_mempool_count_limit(4).with_ready_count(2).build();
+	api.set_nonce(api.genesis_hash(), Bob.into(), 300);
+	api.set_nonce(api.genesis_hash(), Charlie.into(), 400);
+	api.set_nonce(api.genesis_hash(), Dave.into(), 500);
+	api.set_nonce(api.genesis_hash(), Eve.into(), 600);
+	api.set_nonce(api.genesis_hash(), Ferdie.into(), 700);
+
+	let header01 = api.push_block(1, vec![], true);
+	let event = new_best_block_event(&pool, None, header01.hash());
+	block_on(pool.maintain(event));
+
+	let xt0 = uxt(Alice, 200);
+	let xt1 = uxt(Bob, 300);
+	let xt2 = uxt(Charlie, 400);
+
+	let xt3 = uxt(Dave, 500);
+
+	let xt4 = uxt(Eve, 600);
+	let xt5 = uxt(Ferdie, 700);
+
+	api.set_priority(&xt0, 1);
+	api.set_priority(&xt1, 2);
+	api.set_priority(&xt2, 3);
+	api.set_priority(&xt3, 4);
+
+	api.set_priority(&xt4, 5);
+	api.set_priority(&xt5, 6);
+
+	let xt0_watcher = block_on(pool.submit_and_watch(invalid_hash(), SOURCE, xt0.clone())).unwrap();
```

Comment:
nit: a bit misguided about the `invalid_hash()`, `Default::default()` can also do it.

```suggestion
	let xt0_watcher = block_on(pool.submit_and_watch(Default::default(), SOURCE, xt0.clone())).unwrap();
```

Reply:
`deafault` would also do the work, by using `invalid` I wanted to highlight that it is not relevant (param is not used). I also have it in every test, so if we decide to rename, we should do it once in all tests to be consistent.